### ISSUE TITLE
Respond to CTRL+C SIGINT

### ISSUE THREAD

**Issue by lhstrh** [lf-lang/reactor-cpp#43]
@zekailin00 found that when he attempts to interrupt the execution of his DeepRacer controller that uses the Cpp target, the program just keeps on running. Both Ctrl + C and Ctrl + D are ignored, and Ctrl + Z lets the program end abruptly (sometimes triggering a segfault). It is important to free up resources at the end of execution, or else things like GPIO pins become inaccessible during subsequent runs of the program, making debugging very onerous. Can we respond to Ctrl + C by initiating the shutdown sequence, like we're doing in reactor-c?

**Comment by revol-xut** [comment]
Yes there are currently no OS signal handling. We were hesitant to implement this because this means a lot of macro magic to handle different operating systems. But can do

**Comment by lhstrh** [comment]
I think it would be very useful to have it. I think the use case above alone is compelling...
